controllers/usersController.js

```diff
@@ -33,19 +35,24 @@ exports.signup = async (req, res) => {
 
     const avatarURL = gravatar.url(email, { s: '200', r: 'pg', d: 'mm' });
     const hashedPassword = await bcrypt.hash(password, 10);
+    const verificationToken = uuidv4();
     const newUser = new User({
       email,
       password: hashedPassword,
       subscription: "starter",
-      avatarURL
+      avatarURL,
+      verify: false,
+      verificationToken
     });
 
     await newUser.save();
+    mailService.sendVerificationEmail(email, verificationToken); 
```

Comment:
Ta funkcja jest asynchroniczna a tutaj jest tak wpisana jakby była synchroniczna. Staraj się (przynjamniej na początku) używać await i pisać kod bez mieszania stylów. Czyli to też powinno być awaited a cała logika mailservice jak pisałem wcześniej, powinna być napisana w tym stylu też.